mlx_graphs/datasets/dblp.py

```diff
@@ -0,0 +1,141 @@
+    artificial intelligence, information retrieval).
+    Each author is described by a bag-of-words representation of their paper
+    keywords.
+
+    Args:
+        base_dir (str): directory where the dataset should be saved.
+        transform (callable, optional): A function/transform that takes in an
+            :obj:`HeteroGraphData` object and returns a
+            transformed version. The data object will be transformed before
+            every access. (default: :obj:`None`)
+        pre_transform (callable, optional): A function/transform that takes in
+            an `HeteroGraphData` object and returns a
+            transformed version. The data object will be transformed before
+            being saved to disk. (default: :obj:`None`)
+    """
+
+    def __init__(
+        self,
+        base_dir: str,
+        transform: Optional[Callable] = None,
+        pre_transform: Optional[Callable] = None,
+    ):
+        super().__init__(
+            name="DBLP",
+            base_dir=base_dir,
+            transform=transform,
+            pre_transform=pre_transform,
+        )
+
+    @property
+    def raw_path(self) -> str:
+        return f"{super(self.__class__, self).raw_path}"
+
+    @property
+    def raw_file_names(self) -> List[str]:
+        return [
+            "adjM.npz",
+            "features_0.npz",
+            "features_1.npz",
+            "features_2.npz",
+            "labels.npy",
+            "train_val_test_idx.npz",
+        ]
+
+    def download(self):
+        url = "https://www.dropbox.com/s/yh4grpeks87ugr2/DBLP_processed.zip?dl=1"
+        path = download(url=url, path=self.raw_path)
+        new_path = path.split("?")[-2]
+        os.rename(path, new_path)
+        extract_archive(new_path, self.raw_path)
+        os.remove(new_path)
+
+    def process(self):
+        try:
+            import scipy.sparse as sp
+        except ImportError:
+            raise ImportError("scipy is required to download and process the raw data")
+        node_types = ["author", "paper", "term", "conference"]
+        node_features_dict = {}
+        for i, node_type in enumerate(node_types[:2]):
+            nodes = sp.load_npz(osp.join(self.raw_path, f"features_{i}.npz"))
+            node_features_dict[node_type] = mx.array(nodes.todense())
+
+        term = np.load(osp.join(self.raw_path, "features_2.npy"))
+        node_features_dict["term"] = mx.array(term).astype(mx.float32)
+
+        node_type_idx = np.load(osp.join(self.raw_path, "node_types.npy"))
+        node_type_idx = mx.array(node_type_idx).astype(mx.int32)
+
+        """
+        Conference nodes don't have features and hence adding
+        it explicitly to a dictionary will not make sense.
+        Either override the property in the class or set attribute separately
+        for conference
+        """
+        conference_nodes = int((node_type_idx == 3).sum().item())
+        print(conference_nodes)
```

Comment:
@vinayhpandya I'm not sure I understand what's going on here - is `conference_nodes` used anywhere?

Reply:
yes it is used for getting the edge index involving conference nodes.

Reply:
got it 👍